packages/google_maps_flutter/google_maps_flutter_android/example/lib/ground_overlay.dart

```diff
@@ -0,0 +1,325 @@
+  final LatLng _groundOverlayPos1 = const LatLng(37.422026, -122.085329);
+  final LatLng _groundOverlayPos2 = const LatLng(37.42, -122.08);
+  late LatLng _currentGroundOverlayPos;
+
+  // Bounds for demonstranting placing ground overlays with bounds, and
+  // changing bounds.
+  final LatLngBounds _groundOverlayBounds1 = LatLngBounds(
+      southwest: const LatLng(37.42, -122.09),
+      northeast: const LatLng(37.423, -122.084));
+  final LatLngBounds _groundOverlayBounds2 = LatLngBounds(
+      southwest: const LatLng(37.421, -122.091),
+      northeast: const LatLng(37.424, -122.08));
+  late LatLngBounds _currentGroundOverlayBounds;
+
+  Offset _anchor = const Offset(0.5, 0.5);
+
+  Offset _dimensions = const Offset(1000, 1000);
+
+  // Index to be used as identifier for the ground overlay.
+  // If position is changed to bounds and vice versa, the ground overlay will
+  // be removed and added again with the new type. Also anchor can be given only
+  // when the ground overlay is created with position and cannot be changed
+  // after the ground overlay is created.
+  int _groundOverlayIndex = 0;
+
+  @override
+  void initState() {
+    _currentGroundOverlayPos = _groundOverlayPos1;
+    _currentGroundOverlayBounds = _groundOverlayBounds1;
+    super.initState();
+  }
+
+  // ignore: use_setters_to_change_properties
+  void _onMapCreated(ExampleGoogleMapController controller) {
+    this.controller = controller;
+  }
+
+  void _removeGroundOverlay() {
+    setState(() {
+      _groundOverlay = null;
+    });
+  }
+
+  Future<void> _addGroundOverlay() async {
+    final AssetMapBitmap assetMapBitmap = await AssetMapBitmap.create(
+      createLocalImageConfiguration(context),
+      'assets/red_square.png',
+      bitmapScaling: MapBitmapScaling.none,
+    );
+
+    _groundOverlayIndex += 1;
+
+    final GroundOverlayId id =
+        GroundOverlayId('ground_overlay_$_groundOverlayIndex');
+
+    final GroundOverlay groundOverlay = switch (_placingType) {
+      _GroundOverlayPlacing.position => GroundOverlay.fromPosition(
+          groundOverlayId: id,
+          image: assetMapBitmap,
+          position: _currentGroundOverlayPos,
+          width: _dimensions.dx,
+          height: _dimensions.dy,
+          anchor: _anchor,
+          onTap: () {
+            _onGroundOverlayTapped();
+          },
+        ),
+      _GroundOverlayPlacing.bounds => GroundOverlay.fromBounds(
+          groundOverlayId: id,
+          image: assetMapBitmap,
+          bounds: _currentGroundOverlayBounds,
+          anchor: _anchor,
+          onTap: () {
+            _onGroundOverlayTapped();
+          },
+        ),
+    };
+
+    setState(() {
+      _groundOverlay = groundOverlay;
+    });
+  }
+
+  void _onGroundOverlayTapped() {
+    _changePosition();
+  }
+
+  void _setBearing() {
+    assert(_groundOverlay != null);
+    setState(() {
+      _groundOverlay = _groundOverlay!.copyWith(
+          bearingParam: _groundOverlay!.bearing >= 350
```

Comment:
Can you add a comment explaining where these magic numbers come from? 
I believe this is just bumping the bearing 10 degrees  and making sure we dont go over 360 but  that is after reading the whole pr.

Reply:
Is this ok?

Reply:
https://github.com/flutter/packages/pull/8563/files#diff-db70730aaecef994ff31ac3529c7fc25e0a68e0c17c0d2ccba0ae429efc3a2c4R134-R135